File: 198-house-robber/198-house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.size()==1)
        {
            return nums[0];
        }
        else if(nums.size()==2)
        {
            return max(nums[0],nums[1]);
        }
        int i=nums[0];
        int j=max(nums[0],nums[1]);
        int rob=0;        
        for (int k=2;k<nums.size();k++)
        {
            rob = max( i + nums[k] , j );
            i = j;
            j = rob;
        }
        return rob;
        
    }
};

File: 198-house-robber/198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        x=len(nums)
        if x==1:
            return nums[0]
        if x==2:
            return max(nums[0],nums[1])
        t1 = nums[0]
        t2 = max(nums[0],nums[1])
        for i in range(2,x):
            rob = max(nums[i] + t1 , t2)
            t1=t2
            t2=rob
        return rob